compiler/rustc_target/src/spec/mod.rs

```diff
@@ -3394,6 +3395,18 @@ impl Target {
             s => s.clone(),
         }
     }
+
+    pub fn binary_format(&self) -> object::BinaryFormat {
+        if self.is_like_osx {
+            object::BinaryFormat::MachO
+        } else if self.is_like_windows {
+            object::BinaryFormat::Coff
+        } else if self.is_like_aix {
+            object::BinaryFormat::Xcoff
+        } else {
+            object::BinaryFormat::Elf
+        }
+    }
```

Comment:
Maybe this should just be a target spec field in the future with elf as default (as the vast majority of the targets uses elf), but for now this is completely fine.

Reply:
Probably worth leaving this as a comment, or some enhancement issue.

Reply:
#135724